lcdb/reporting.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+""" Reporting System """
+
+import os
+import sys
+import base64
+from textwrap import dedent
+
+import markdown
+import jinja2
+
+TEMPLATES_DIR = os.path.join(os.path.dirname(__file__), 'templates')
+loader = jinja2.FileSystemLoader(TEMPLATES_DIR)
+ENV = jinja2.Environment(loader=loader)
+
+def report(string, **kwargs):
+    """ Build a report """
+    md = markdown.Markdown(output_format='html5')
+    tp = ENV.get_template('base.html')
+    return tp.render(content=md.convert(dedent(string).format(**kwargs)))
+
```

Comment:
Yes, I think the cleanest/easiest way is to just have a rule for each HTML blob and then use the report rule to write the text and insert the blobs in the correct order.